crates/storage/db/src/lib.rs

```diff
@@ -171,26 +172,70 @@ pub mod test_utils {
     /// Error during tempdir creation
     pub const ERROR_TEMPDIR: &str = "Not able to create a temporary directory.";
 
+    /// A database will delete the db dir when dropped.
+    #[derive(Debug)]
+    pub struct TestTempDataBase<DB> {
+        db: Option<DB>,
```

Comment:
because the `drop` function need take the `db` out and drop it first to close the database, or else the `std::fs::remove_dir_all(&self.path)` will fail .